File: 242_Valid_Anagram/main.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        record = [0] * 26
        for char in s:
            ctoi = ord(char) - 97
            record[ctoi] += 1

        for char in t:
            ctoi = ord(char) - 97
            record[ctoi] -= 1
            if record[ctoi] < 0:
                return False

        return True

File: 242_Valid_Anagram/main.go
func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}
	record := make(map[rune]int)

	runes := []rune(s)
	for i := 0; i < len(runes); i++ {
		record[runes[i]]++
	}

	runes = []rune(t)
	for i := 0; i < len(runes); i++ {
		record[runes[i]]--
		if record[runes[i]] < 0 {
			return false
		}
	}

	return true
}
